CMaxTable.lua

```diff
@@ -19,8 +19,7 @@ end
 
 function CMaxTable:updateGradInput(input, gradOutput)
    for i=1,#input do
-      self.gradInput[i] = torch.Tensor()
-      self.gradInput[i]:resizeAs(input[i]):fill(0.0)
+      self.gradInput[i] = input[i]:clone():fill(0.0)
       local mask = torch.eq(self.maxIdx, i)
```

Comment:
Just a heads up that this allocates memory at every iteration. With current GPU memory allocator, this slows down things quite a lot. @soumith should we care about it ?